libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -606,6 +605,7 @@ class MapboxNavigation @VisibleForTesting internal constructor(
     @RequiresPermission(anyOf = [ACCESS_COARSE_LOCATION, ACCESS_FINE_LOCATION])
     @JvmOverloads
     fun startTripSession(withForegroundService: Boolean = true) {
+        threadController.assertSDKThread()
```

Comment:
I think the UiThread annotation should be removed from class definition. It may trigger some lint errors.

Reply:
fair enough